packages/server/test/orchestrator.test.ts

```diff
@@ -75,46 +43,165 @@ describe('orchestrator', () => {
     });
 
     it('serves the application', () => {
-      expect(body).toContain('<title>React TodoMVC Example</title>');
+      expect(body).toContain('Signing In');
     });
   });
 
-  describe('retrieving app via proxy', () => {
-    let response: Response;
-    let body: string;
+  describe('an unreachable application', () => {
     beforeEach(async () => {
-      response = await actions.fetch('http://localhost:24101/');
-      body = await response.text();
-    });
-
-    it('responds successfully', () => {
-      expect(response.ok).toEqual(true);
-    });
-
-    it('proxies to the application', () => {
-      expect(body).toContain('<title>React TodoMVC Example</title>');
-    });
-
-    it('injects the harness script tag', () => {
-      expect(body).toMatch(new RegExp(`<script src="http://localhost:\\d+/__bigtest/harness.js"></script>`, 'mg'));
+      await actions.fork(function * () {
+        yield actions.atom.once(state => state.appService.appStatus === 'unreachable')
+      });
+    });
+    
+    it ('indicates that the application is unreachable', () => {
+      expect(actions.atom.get().appService.appStatus).toEqual('unreachable');
+    });
+  
+    describe('retrieving app via proxy', () => {
+      let response: Response;
+
+      beforeEach(async () => {
+        response = await actions.fetch('http://localhost:24101/');
+      });
+  
+      it('responds with a server error', () => {
+        expect(response.status).toEqual(502);
+      });
     });
   });
 
-  describe('retrieving test file manifest', () => {
-    let response: Response;
-    let body: string;
-    beforeEach(async () => {
-      let name = actions.atom.get().manifest.fileName;
-      response = await actions.fetch(`http://localhost:24105/${name}`);
-      body = await response.text();
-    });
-
-    it('responds successfully', () => {
-      expect(response.ok).toEqual(true);
+  describe('an externally managed application', () => {
+    beforeEach(async function() {
+      await actions.fork(function * () {
+        let child = yield ChildProcess.spawn("yarn test:app:start 24100");
+
+        actions.registerAfterDestroyPromise(
+          new Promise(resolve => {
+            child.on('exit', () => setTimeout(resolve, 100));
```

Comment:
@cowboyd - waiting for the child to exit wasn't enough to solve the port collision issue, based on experimentation I had to wait longer for the port to be available.

Reply:
as mentioned elsewhere, what about just starting on a fresh port each time?